src/visualizr/anitalker/experiment.py

```
@@ -340,0 +1,343 @@
+        self.warmup = warmup
+
+    def __call__(self, step):
+        return min(step, self.warmup) / self.warmup
+
+
+def is_time(num_samples, every, step_size):
+    closest = (num_samples // every) * every
+    return num_samples - closest < step_size
+
+
+def train(conf: TrainConfig, gpus, nodes=1):
```

Ruff (0.12.2)

Missing docstring in public function

(D103)